### PR TITLE
Added override cycles logic

### PR DESCRIPTION
**Pull request by alexiswl**
--no-override-cycles parameter to stop this logic from processing.

Otherwise, by default adds Override cycles to [Settings] block in SampleSheet.csv.

If the index has trailing Ns, the index is stripped to that length and the rest is populated with Ns,
i.e
an index of length 8 when 10 cycles have been run will present as:
i8n2

Reads are taken from the [Reads] block for R1 and R2.

Also fixed bug where index_len and index2_len columns were being added to the output samplesheet.csv file.

### CODE REVIEW

File: scripts/showcase/samplesheet-prep-test.py [hunk 1]
```diff
@@ -227,12 +239,38 @@ def convert_sample_sheet_header_to_v2(sample_sheet_header_rows):
...
+            if not index2_len == index2_len_orig:
+                # We need to mask the length of the difference
+                i5_mask = "{}i{}n".format(index2_len, index2_len_orig-index2_len)
+            else:
+                i5_mask = "{}i".format(index2_len)
+
+            sample_sheet_header_rows_new.append("OverrideCycles,{};{};{};{}\n".format(
+                r1_mask, i7_mask, i5_mask, r2_mask
+            ))
         sample_sheet_header_rows_new.append(new_row)
```
Comment (skanwal) [inline scripts/showcase/samplesheet-prep-test.py:272]: An example of a correct format for overridecycles setting for bclConvert is:
`OverrideCycles,Y151;I8N2;I8N2;Y151`

You are adding:
OverrideCycles,151y;8i2n;8i2n;151y

Comment (alexiswl) [reply scripts/showcase/samplesheet-prep-test.py:272]: :facepalm:

Comment (alexiswl) [reply scripts/showcase/samplesheet-prep-test.py]: This should be fixed now with #57

**Comment by alexiswl** [comment]
@skanwal can we merge this.

**Comment by skanwal** [comment]
Sorry - forgot to press the button after testing. Thanks.

**Pull request merged**